### PR TITLE
xray: add xray module and grant xray permissions to ecs and stack 

### PR DESCRIPTION
**Pull request by samkahchiin**
#### Summary
Add xray module and grant xray permissions to ECS and stack

### CODE REVIEW

File: stack/app/README.md [hunk 1]
```diff
@@ -114,6 +114,8 @@ module "stack" {
   secret_arns                     = []
   # appends region to the name (usually ${project}-${environment}) for globally unique names
   regional                        = true
+  enable_xray                     = true
```
Comment (swiknaba) [inline stack/app/README.md:117]:
```suggestion
  enable_xray                     = false
```

should we put the actual default value here?

**Review by swiknaba — APPROVED** [review]
Great thanks!

File: xray/xray-sampling.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Refer to here for the sampling rules
```
Comment (swiknaba) [inline xray/xray-sampling.tf:1]: what do we need this for?

**Review by swiknaba — APPROVED** [review]
LGTM!

File: xray/xray-sampling.tf (hunk 2 above)
Comment (samkahchiin) [reply xray/xray-sampling.tf:1]: Do u mean the comments or the files?

Comment (samkahchiin) [reply xray/xray-sampling.tf:1]: Do you refer to the comments? 
Leave a note here so we have a docs reference of the sampling rules. Not a really important notes actually

Comment (swiknaba) [reply xray/xray-sampling.tf:1]: ah, so this is excluding traces for the health check endpoints, good thinking!

**Pull request merged**